### PR TITLE
Add playbooks linter

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
In this PR I'm adding a linter to check if each alert is referenced in playbooks. Since we have several alerts without a playbook, I've temporarily added them to `playbooks.md` to let the linter pass (and will progressively address them in follow up PRs). However, the new linter will enforce that any new alert will have the playbook too starting from now.

Example of failed playbooks linter:
https://app.circleci.com/pipelines/github/grafana/cortex-jsonnet/852/workflows/8932fcad-3009-45e7-9124-1427816cb319/jobs/2115

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: build-image/Dockerfile [hunk 1]
```diff
@@ -27,12 +27,17 @@ RUN chmod +x /usr/bin/tk
 
 # Build mixtool
 FROM golang:1.15-alpine AS mixtool-builder
-RUN GO111MODULE=on go get github.com/monitoring-mixins/mixtool/cmd/mixtool@59d44357240d
```
Comment (pracucci) [inline build-image/Dockerfile:30]: Alerts and rules generation didn't work with the previous version 😱

File: Makefile [hunk 2]
```diff
@@ -2,19 +2,26 @@
...
+lint-mixin: lint-mixin-with-mixtool lint-mixin-with-jsonnetfmt
+
+lint-mixin-with-jsonnetfmt:
 	@RESULT=0; \
 	for f in $$(find . -name '*.libsonnet' -print -o -name '*.jsonnet' -print); do \
 		$(JSONNET_FMT) -- "$$f" | diff -u "$$f" -; \
 		RESULT=$$(($$RESULT + $$?)); \
 	done; \
-	RESULT=$$(($$RESULT + $$?)); \
+	exit $$RESULT
```
Comment (pracucci) [inline Makefile:15]: This lint check was failing (see changes to `scaling.libsonnet`) but wasn't reported by CI cause the status code was 0.

File: scripts/lint-playbooks.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  echo "No alerts found. Something went wrong with the listing."
+  exit 1
+fi
+
+# Check if each alert is referenced in the playbooks.
+STATUS=0
+
+for ALERT in $ALERTS; do
+  grep -q "$ALERT" "${SCRIPT_DIR}/../cortex-mixin/docs/playbooks.md"
+  if [ $? -ne 0 ]; then
```
Comment (pstibrany) [inline scripts/lint-playbooks.sh:20]: nit:

```suggestion
  if ! grep -q "$ALERT" "${SCRIPT_DIR}/../cortex-mixin/docs/playbooks.md"; then 
```

**Review by pstibrany — APPROVED** [review]
Nice work!

**Pull request merged**